isis/src/base/apps/isis2pds/isis2pds.cpp

```diff
@@ -0,0 +1,244 @@
+#include <QString>
+
+#include "Cube.h"
+#include "ExportDescription.h"
+#include "FileName.h"
+#include "Process.h"
+#include "ProcessExportPds.h"
+#include "ProcessExportPds4.h"
+#include "Pvl.h"
+#include "PvlKeyword.h"
+#include "PvlToXmlTranslationManager.h"
+#include "isis2pds.h"
+
+using namespace std;
+namespace Isis{
+
+  enum Pixtype { NONE, NEG, BOTH };
+
+  void setRangeAndPixels(UserInterface &ui, ProcessExport &p,
+                         double &min, double &max, Pixtype ptype);
+  void isis2pds(UserInterface &ui, Pvl *log) {
+
+    // Check if input file is indeed, a cube
+    if (ui.GetFileName("FROM").right(3) != "cub") {
+      QString msg = "Input file [" + ui.GetFileName("FROM") +
+                  "] does not appear to be a cube";
+      throw  IException(IException::User, msg, _FILEINFO_);
+    }
```

Comment:
I know you didn't add these lines, but I think they are unreachable. If I remember correctly, the IsisAml class would throw an error if the input wasn't a cube. Is this right @scsides?

Reply:
I understand what you mean, but the code is reachable because the Aml does not check the file name or open it and make sure it is a cube prior to calling IsisMain.

Along the same lines, it is an odd test. Why not just wait until the file is opened with "open" or "setInputCube"? Either of those will fail gracefully and accurately if it is not a cube.

Reply:
Right, I see what you mean. It looks like in the cube class, it will try to append the cub extension to the input file. With these lines in isis2pds, we remove that additional functionality in the cube class.

Reply:
So it sounds like overall this code block is not necessary?